### ISSUE TITLE
Improve operation history robustness by fixing Json.

### ISSUE THREAD

**Issue by GoogleCodeExporter** [Illablu/google-refine#88]
```
When applying operations, try to fix common json errors.  We should expect that 
developers of gridworks or extensions may make errors when providing Json 
history for operations.

Gridworks should try to be more robust and attempt to fix missing or extraneous 
commas and missing top level brackets.  It should report any fixes made or 
errors which it was unable to fix.
```

Original issue reported on code.google.com by `iainsproat` on 23 Jun 2010 at 4:03

**Comment by GoogleCodeExporter** [comment]
```
I'm not so sure that this is a good idea. Making the Json parser more tolerant 
might just remove the incentive for extension developers to fix their code (or 
extension users to upgrade). 

It reminds me of how early web browsers supported malformed HTML to try and 
make the web "just work". We're still dealing with the aftermath of that 
decision.
```

Original comment by `narphor...@gmail.com` on 23 Jun 2010 at 4:13

**Comment by GoogleCodeExporter** [comment]
```
This was actually fixed a while back for 2.0. It just fixed missing commas or [ 
], which are easy to miss when editing a list of operations.
```

Original comment by `dfhu...@gmail.com` on 14 Aug 2011 at 3:02
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
Original comment by `tfmorris` on 18 Sep 2012 at 3:03
- Added labels: **Milestone-2.5**
